src/modules/jobs/job.service.ts

```diff
@@ -33,33 +37,103 @@ export class JobsService {
     }
   }
 
-  async addJobToUser(
-    owner: User,
-    payload: AddJobDescriptionDto,
-  ): Promise<void> {
+  async findJobs(params: GetJobsDto): Promise<FindJobsResponse> {
     try {
-      await this.jobRepository
-        .createQueryBuilder()
-        .insert()
-        .into(Job)
-        .values([{ ...payload, owner }])
-        .execute();
-    } catch (error) {
-      if (error instanceof HttpException) {
-        throw error;
+      const {
+        limit,
+        offset,
+        skills,
+        categories,
+        hourly_rate_start,
+        hourly_rate_end,
+        search,
+        ...jobPayload
+      } = params;
+
+      const query = this.jobRepository
+        .createQueryBuilder('job')
+        .leftJoinAndSelect('job.owner', 'user')
+        .leftJoinAndSelect('job.category', 'category')
+        .where(jobPayload)
+        .limit(params.limit || findJobsDefaultLimit)
+        .offset(params.offset || findJobsDefaultOffset)
+        .orderBy('job.created_at', 'DESC');
+
+      if (hourly_rate_start) {
+        query.andWhere('job.hourly_rate >= :hourly_rate_start', {
+          hourly_rate_start,
+        });
+      }
+
+      if (hourly_rate_end) {
+        query.andWhere('job.hourly_rate <= :hourly_rate_end', {
+          hourly_rate_end,
+        });
+      }
+
+      if (search) {
+        query.andWhere(
+          new Brackets((qb) => {
+            qb.where('job.title LIKE :search').orWhere(
+              'job.description LIKE :search',
+              {
+                search: `%${search}%`,
+              },
+            );
+          }),
+        );
       }
+
+      if (categories) {
+        query.andWhere('job.category.id IN (:...categories)', { categories });
+      }
+
+      if (skills) {
+        query.innerJoin('job.skills', 'skill', 'skill.id IN (:...skills)', {
+          skills,
+        });
+      }
+
+      const [jobs, totalCount] = await query.getManyAndCount();
+
+      return {
+        jobs,
+        meta: {
+          totalCount,
+        },
+      };
+    } catch (error) {
       throw new InternalServerErrorException();
     }
   }
 
-  async addJobInfo(
-    payload: AddJobDescriptionDto,
-    user: UserDto,
-  ): Promise<void> {
+  async createJob(payload: CreateJobDto, user: UserDto): Promise<void> {
     try {
-      const currentUser = await this.userService.findByEmail(user.email);
+      const owner = await this.userService.findByEmail(user.email);
```

Comment:
find other way, you already have `user`, its expensive to make query to get same data

Reply:
@lyudad fixed